packages-web/datagrid/src/Datagrid.tsx

```diff
@@ -37,20 +37,21 @@ export default function Datagrid(props: DatagridContainerProps): ReactElement {
         <Table
             className={props.class}
             columns={props.columns}
-            data={props.datasource.items ?? []}
-            setPage={setPage}
-            hasMoreItems={props.datasource.hasMoreItems ?? false}
-            page={currentPage}
             columnsDraggable={props.columnsDraggable}
             columnsFilterable={props.columnsFilterable}
+            columnsHidable={props.columnsHidable}
             columnsResizable={props.columnsResizable}
             columnsSortable={props.columnsSortable}
-            columnsHidable={props.columnsHidable}
+            data={props.datasource.items ?? []}
+            footerWidgets={<div className="header">{props.footerWidgets}</div>}
+            hasMoreItems={props.datasource.hasMoreItems ?? false}
```

Comment:
You can actually compute `numberOfPages` from `datasource.totalCount` if it is available

Reply:
Done

Reply:
This was amazing! Now we have `Got to last page` even for some cases of BE paging